packages/kbn-search-connectors/lib/update_native.ts

```diff
@@ -6,24 +6,35 @@
  * Side Public License, v 1.
  */
 
-import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { Result } from '@elastic/elasticsearch/lib/api/types';
 
-import { CONNECTORS_INDEX } from '..';
-import { Connector, ConnectorStatus } from '../types/connectors';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { i18n } from '@kbn/i18n';
+import { isNotFoundException } from '../utils/identify_exceptions';
 
 export const putUpdateNative = async (
   client: ElasticsearchClient,
   connectorId: string,
   isNative: boolean
 ) => {
-  const result = await client.update<Connector>({
-    doc: {
-      is_native: isNative,
-      status: ConnectorStatus.CONFIGURED,
-    },
-    id: connectorId,
-    index: CONNECTORS_INDEX,
-  });
+  try {
+    const result = await client.transport.request<Result>({
+      method: 'PUT',
+      path: `/_connector/${connectorId}/_native`,
+      body: {
+        is_native: isNative,
+      },
+    });
+    return result;
+  } catch (err) {
```

Comment:
We should say "could not find connector" if we're going to rethrow errors. "Document" is not informative enough.

Reply:
But ideally we just pass through the Elasticsearch error, there's really no need to catch and rethrow here.

Reply:
Ack, yeah I agree that errors from ES are descriptive and should be enough to figure out what's wrong.